flax/experimental/nnx/nnx/transforms.py

```diff
@@ -111,26 +113,116 @@ def check_and_call(accessor: DelayedAccessor, *args, **kwargs):
 
 UNSPECIFIED = object()
 
+@dataclasses.dataclass(frozen=True)
+class JitStaticInputs:
+  graphdef: GraphDef[tuple[tp.Any, ...]]
+
+
+jax.tree_util.register_static(JitStaticInputs)
+
+
+@dataclasses.dataclass(frozen=True)
+class JitStaticOutputs:
+  graphdef: GraphDef[tuple[tp.Any, ...]]
+  index_mapping: dict[Index, Index]
+
+
+jax.tree_util.register_static(JitStaticOutputs)
+
 
 @dataclasses.dataclass
 class JITOptions:
   in_shardings: tp.Any
   out_shardings: tp.Any
-  static_argnums: tp.Union[int, tp.Sequence[int], None]
-  static_argnames: tp.Union[str, tp.Iterable[str], None]
-  donate_argnums: tp.Union[int, tp.Sequence[int]]
+  static_argnums: tuple[int, ...]
+  static_argnames: tuple[str, ...]
+  donate_argnums: tuple[int, ...]
+  donate_argnames: tuple[str, ...]
   keep_unused: bool
   device: tp.Optional[jax.Device]
   backend: tp.Optional[str]
   inline: bool
   abstracted_axes: tp.Optional[tp.Any]
+  donate_object_state: bool
+
+  @classmethod
+  def from_jit_kwargs(
+    cls,
+    in_shardings: tp.Any,
+    out_shardings: tp.Any,
+    static_argnums: int | tp.Sequence[int] | None,
```

Comment:
Is there a reason why `static_argunums` is a `Sequence`, but `static_argnames` can be an `Iterable`?

Reply:
I copied this form `jax.jit`, maybe it has to support repeated iteration so they request sequence?